terragrunt/aws/glue/etl/platform/gc_forms/process_data.py

```diff
@@ -204,63 +248,93 @@ def publish_metric(cloudwatch, dataset_name, count, processing_time):
     )
 
 
-def process_data():
+def download_s3_object(s3, s3_url, filename):
+    """
+    Download an S3 object to a local file.
+    """
+    bucket_name = s3_url.split("/")[2]
+    object_key = "/".join(s3_url.split("/")[3:])
+    current_dir = os.getcwd()
+    s3.download_file(
+        Bucket=bucket_name,
+        Key=object_key,
+        Filename=os.path.join(current_dir, filename),
+    )
+
+    gx_zip = os.path.join(current_dir, os.path.basename(filename))
+    gx_dir = os.path.join(current_dir, os.path.splitext(gx_zip)[0])
+
+    with zipfile.ZipFile(gx_zip, "r") as zip_ref:
+        zip_ref.extractall(gx_dir)
+
+
+def process_data(datasets=None):
     """
     Main ETL process to read data from S3, validate the schema, and save the
     transformed data back to S3.
     """
-    datasets = [
-        {
-            "path": "historical-data",
-            "date_columns": ["date"],
-            "partition_timestamp": "date",
-            "partition_columns": ["year", "month"],
-            "email_columns": ["client_email"],
-        },
-        {
-            "path": "processed-data/submissions",
-            "date_columns": ["timestamp"],
-        },
-        {
-            "path": "processed-data/template",
-            "date_columns": [
-                "ttl",
-                "api_created_at",
-                "timestamp",
-                "closingdate",
-                "created_at",
-                "updated_at",
-            ],
-            "field_count_columns": [
-                "checkbox_count",
-                "combobox_count",
-                "dropdown_count",
-                "dynamicrow_count",
-                "fileinput_count",
-                "formatteddate_count",
-                "radio_count",
-                "richtext_count",
-                "textarea_count",
-                "textfield_count",
-                "addresscomplete_count",
-            ],
-            "partition_timestamp": "created_at",
-            "partition_columns": ["year", "month"],
-            "email_columns": ["deliveryemaildestination"],
-        },
-        {
-            "path": "processed-data/templateToUser",
-            "date_columns": ["timestamp"],
-        },
-        {
-            "path": "processed-data/user",
-            "date_columns": ["emailverified", "lastlogin", "createdat", "timestamp"],
-            "partition_timestamp": "lastlogin",  # User created date is currently in this field.
-            "partition_columns": ["year", "month"],
-            "drop_columns": ["name"],
-            "email_columns": ["email"],
-        },
-    ]
+    s3 = boto3.client("s3")
+    download_s3_object(s3, GX_CONFIG_OBJECT, "tables.zip")
```

Comment:
Minor tweak here just to match up filenames.  I honestly don't think it matters since when we unzip it I think we'll get the folder named `gx` anyway:
```suggestion
    download_s3_object(s3, GX_CONFIG_OBJECT, "gx.zip")
```